fitbenchmarking/parsing/tests/test_parsers.py

```diff
@@ -0,0 +1,235 @@
+"""
+This file contains tests for the parsers.
+"""
+
+from importlib import import_module
+from inspect import isclass, isabstract, getmembers
+from json import load
+import numpy as np
+import os
+
+from fitbenchmarking.parsing.base_parser import Parser
+from fitbenchmarking.parsing.fitting_problem import FittingProblem
+from fitbenchmarking.parsing.parser_factory import ParserFactory
+
+
+def pytest_generate_tests(metafunc):
+    """
+    Function used by pytest to parametrize tests.
+    This will create a set of tests for each function in the class where
+    the parameters are given in a 'params' dict in the class.
+    """
+    # called once per each test function
+    funcarglist = metafunc.cls.params[metafunc.function.__name__]
+    argnames = sorted(funcarglist[0])
+    argvals = [[funcargs[name]
+                for name in argnames]
+               for funcargs in funcarglist]
+    metafunc.parametrize(argnames,
+                         argvals)
+
+
+def generate_test_cases():
+    """
+    Utility function to create the params dict for parametrising the tests.
+
+    :return: params dictionary with a function name as a key, and a list of
+             parameter dictionaries for the value
+    :rtype: dict
+    """
+    params = {'test_parsers': [],
+              'test_factory': [],
+              'test_function_evaluation': []}
+
+    # get all parsers
+    test_dir = os.path.dirname(__file__)
+    formats = [f[:-10] for f in os.listdir(os.path.join(test_dir, os.pardir))
+               if f.endswith('_parser.py')
+               and f != 'base_parser.py']
+
+    # create list of test_cases
+    expected_dir = os.listdir(os.path.join(test_dir, 'expected'))
+    for file_format in formats:
+        format_dir = os.listdir(os.path.join(test_dir, file_format))
+        for expected_file in expected_dir:
+            expected_path = os.path.join(test_dir, 'expected', expected_file)
+            test_name = os.path.splitext(expected_file)[0]
+            test_name_with_ext = [f for f in format_dir
+                                  if f.startswith(test_name)]
+            if len(test_name_with_ext) == 0:
```

Comment:
```suggestion
            if test_name_with_ext:
```
PEP 8 prefers using the fact that empty lists are `False` (it is also marginally more efficient).

Reply:
I didn't know that (and think I prefer explicit, although maybe should have had a `!= []`). I'll update to suite PEP8 though.

(Can't accept the suggestion as I think it flips the logic)

Reply:
Yeah I personally don't really have a strong preference, I think either is fine.  Apologies, should have been more careful with the suggestions.